### DISCUSSION TITLE
[nuxt] [setupSFC] [IDEAS] setupSFC auto imports

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [vue-macros/vue-macros#449] [Ideas]
Now that setupSFC is working in Nuxt (see https://github.com/vue-macros/vue-macros/discussions/429), it opened a window of opportunity for improvements.

Allowing files in  `~/components` and `~/pages` to be auto imported.

For now, we have to import them manually with `import Component from '~/components/component.setup'`.

Explicit import:
![image](https://github.com/vue-macros/vue-macros/assets/10191085/8805ca95-bb83-4a2f-aca9-9cf0fefe383c)

Auto import:
![image](https://github.com/vue-macros/vue-macros/assets/10191085/0faade54-34ff-4834-b47d-b359d66301cf)

### Final Words

~~I don't think we need auto imports for `~/layouts`, because these are imported using `<NuxtLayout name="some layout"/>`.~~
Layouts are not working! I added `~/layouts/main.setup.tsx`, then in `~/pages/index.setup.tsx`:

```tsx
export default () => (
  <NuxtLayout name="main"> // also tried with "main.setup"
     <h1>Index</h1>
    <Working />
  </NuxtLayout>
)
```

Error:

![image](https://github.com/vue-macros/vue-macros/assets/10191085/29a245c8-49be-4db0-980c-243d1c02291e)

Props to the @vue-macros team for the great job! 😍🔥🎉

**Comment by sxzz** [comment]
Auto-import components are not working in JSX, even the normal SFC files (`.vue`). Maybe we should report it to Nuxt core first.

**Comment by sxzz** [comment]
The use of setupSFC in layouts is supported in `@vue-macros/nuxt@1.7.0`.

**Reply by sxzz** [reply]
Demo: https://stackblitz.com/edit/nuxt-starter-crjsmy?file=layouts%2Fmain.setup.tsx
